### PR TITLE
Added support for choosing encapsulation type

### PR DESCRIPTION
**Pull request by shaharyarn**
Related python PR: https://github.com/tomlegkov/marine-python/pull/27

### CODE REVIEW

File: marine.c [hunk 1]
```diff
@@ -184,6 +184,9 @@ static frame_data prev_cap_frame;
 static guint32 epan_auto_reset_count = 20000; // TODO make this configurable
 static gboolean epan_auto_reset = TRUE;
 
+const unsigned int ethernet_encap = 1;
```
Comment (yehonatanz) [inline marine.c]: CAPITAL_LETTERS

File: marine.h [hunk 2]
```diff
@@ -12,10 +12,13 @@ typedef struct {
 
 int init_marine(void);
 void set_epan_auto_reset_count(guint32 auto_reset_count);
-marine_result *marine_dissect_packet(int filter_id, unsigned char *data, int len);
+marine_result *marine_dissect_packet(int filter_id, unsigned char *data, int len, unsigned int encap);
```
Comment (yehonatanz) [inline marine.h]: Doesn't it make more sense for each filter to have an encap?

File: marine_benchmark.c [hunk 3]
```diff
@@ -148,13 +149,13 @@ int main(int argc, char *argv[]) {
...
 
     benchmark_case cases[] = {
-            {"Benchmark with BPF",                                            bpf, NULL,    NULL},
-            {"Benchmark with Display filter",         NULL,                        dfilter, NULL},
-            {"Benchmark with BPF and Display filter",                         bpf, dfilter, NULL},
-            {"Benchmark with three extracted fields", NULL,                        NULL,    three_fields},
-            {"Benchmark with eight extracted fields", NULL,                        NULL,    eight_fields},
-            {"Benchmark with BPF, Display filter and three extracted fields", bpf, dfilter, three_fields},
-            {"Benchmark with BPF, Display filter and eight extracted fields", bpf, dfilter, eight_fields},
+            {"Benchmark with BPF",                                            bpf, NULL,    NULL, 0},
```
Comment (yehonatanz) [inline marine_benchmark.c]: Copy these changes from disable parsing options pr

**Review by yehonatanz — CHANGES_REQUESTED** [review]
Don't forget BPF support!

File: marine.c [hunk 4]
```diff
@@ -184,6 +185,9 @@ static frame_data prev_cap_frame;
 static guint32 epan_auto_reset_count = 20000; // TODO make this configurable
 static gboolean epan_auto_reset = TRUE;
 
+const unsigned int ETHERNET_ENCAP = 1;
```
Comment (yehonatanz) [inline marine.c:188]: Can't you just declare and assign in `marine.h` and remove these lines from `marine.c`?
Seems nicer to me

**Review by yehonatanz — CHANGES_REQUESTED** [review]

File: marine.c (hunk 4 above)
Comment (shaharyarn) [reply marine.c:188]: You cannot, C doesn't let you put an initializer for an extern variable (and you do need to define it as extern in `marine.h`)

File: marine_benchmark.c [hunk 5]
```diff
@@ -148,13 +147,13 @@ int main(int argc, char *argv[]) {
     };
 
     benchmark_case cases[] = {
```
Comment (tomlegkov) [inline marine_benchmark.c]: Maybe add benchmark cases with another encapsulation to see how it affects memory and speed?
Ideally you should run all of the cases with ethernet and then all of the cases with another encapsulation type.

However, I believe this is optional and up to @yehonatanz to decide :)

**Review by tomlegkov — APPROVED** [review]

File: marine_benchmark.c (hunk 5 above)
Comment (yehonatanz) [reply marine_benchmark.c]: I agree, it would be the best but definitely optional

**Review by yehonatanz — APPROVED** [review]

File: cap_maker.py [hunk 6]
```diff
@@ -22,44 +24,63 @@
...
+            + llc.LLC(
+                dsap=170, ssap=170, ctrl=3, snap=int.to_bytes(llc.LLC_TYPE_IP, 5, "big")
+            )
+    )
+    for _ in range(CONVERSATIONS // 2):
+        packets.extend(create_conversation(radiotap_base, tcp.TCP))
+        packets.extend(create_conversation(radiotap_base, udp.UDP))
+
+    random.shuffle(packets)
+    write_cap("radiotap_benchmark.cap", packets, RADIOTAP_PCAP_HEADER)
```
Comment (yehonatanz) [inline cap_maker.py]: Extract all this to `create_cap` function or something

File: marine_benchmark.c [hunk 7]
```diff
@@ -110,7 +110,8 @@ int print_title(char *str) {
 
 int main(int argc, char *argv[]) {
     if (argc < 2) {
-        printf("Usage: ./marine_benchmark <path to cap generated by cap_maker.py>\n");
+        printf("Usage: ./marine_benchmark <path to cap generated by cap_maker.py> "
+               "<encapsulation type value (defaults to ethernet)\n");
```
Comment (yehonatanz) [inline marine_benchmark.c]: You're missing a `>` here

File: marine_benchmark.c [hunk 8]
```diff
@@ -123,6 +124,17 @@ int main(int argc, char *argv[]) {
         return -1;
     }
 
+    int encap_type = -1;
+    if (argc > 2) {
+        sscanf(argv[2], "%d", &encap_type);
+    } else {
+        encap_type = 1;
```
Comment (yehonatanz) [inline marine_benchmark.c]: `ETHERNET_ENCAP`?

**Review by yehonatanz — CHANGES_REQUESTED** [review]
Just those tiny changes and :)

**Review by yehonatanz — APPROVED** [review]

File: cap_maker.py [hunk 9]
```diff
@@ -74,3 +104,6 @@ def create_conversation(protocol: Union[Type[tcp.TCP], Type[udp.UDP]]) -> Iterat
     )
     print("BPF:", bpf)
     print("Display filter:", display_filter)
+    print(
+        "Ethernet cap encapsulation type value: 1\nRadiotap cap encapsulation type value: 23"
```
Comment (tomlegkov) [inline cap_maker.py]: Much better to have two separate prints or multi-line string

**Review by tomlegkov — APPROVED** [review]

**Pull request merged**